src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests.cs

```diff
@@ -9202,6 +9493,356 @@ public UnmanagedCallersOnlyAttribute()
+public class C
+{
+    [A<int>]
+    public void M() { }
+}";
+
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics();
+            CompileAndVerify(comp, symbolValidator: validate, sourceSymbolValidator: validate);
+
+            static void validate(ModuleSymbol module)
+            {
+                var m = (MethodSymbol)module.ContainingAssembly.GlobalNamespace.GetMember("C.M");
+                var attribute = m.GetAttributes().Single();
+                Assert.Equal("A<System.Int32>", attribute.AttributeClass.ToTestDisplayString());
+                Assert.Equal("A<System.Int32>..ctor()", attribute.AttributeConstructor.ToTestDisplayString());
+            }
+        }
+
+        [Fact]
+        public void AmbiguityWithGenericAttribute()
+        {
+            var source = @"
+public class AAttribute<T> : System.Attribute {}
+public class A<T> : System.Attribute {}
+
+[A<int>]
+public class C
+{
+}";
+
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics(
+                // (5,2): error CS1614: 'A<>' is ambiguous between 'A<T>' and 'AAttribute<T>'; use either '@A<>' or 'A<>Attribute'
+                // [A<int>]
+                Diagnostic(ErrorCode.ERR_AmbiguousAttribute, "A<int>").WithArguments("A<>", "A<T>", "AAttribute<T>").WithLocation(5, 2)
+                );
+
+            comp = CreateCompilation(source, parseOptions: TestOptions.Regular9);
+            comp.VerifyDiagnostics(
+                // (2,30): error CS8773: Feature 'generic attributes' is not available in C# 9.0. Please use language version 10.0 or greater.
+                // public class AAttribute<T> : System.Attribute {}
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion9, "System.Attribute").WithArguments("generic attributes", "10.0").WithLocation(2, 30),
+                // (3,21): error CS8773: Feature 'generic attributes' is not available in C# 9.0. Please use language version 10.0 or greater.
+                // public class A<T> : System.Attribute {}
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion9, "System.Attribute").WithArguments("generic attributes", "10.0").WithLocation(3, 21),
+                // (5,2): error CS1614: 'A<>' is ambiguous between 'A<T>' and 'AAttribute<T>'; either use '@A<>' or explicitly include the 'Attribute' suffix
+                // [A<int>]
+                Diagnostic(ErrorCode.ERR_AmbiguousAttribute, "A<int>").WithArguments("A<>", "A<T>", "AAttribute<T>").WithLocation(5, 2),
+                // (5,2): error CS8773: Feature 'generic attributes' is not available in C# 9.0. Please use language version 10.0 or greater.
+                // [A<int>]
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion9, "A<int>").WithArguments("generic attributes", "10.0").WithLocation(5, 2)
+                );
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Consider testing the two recommendations from the ambiguity error:
```
[A<int>]
[AAttribute<int>]
```

Reply:
Adding the test. This revealed bug #54772.